packages/nx/src/project-graph/build-project-graph.ts

```diff
@@ -185,26 +182,20 @@ async function buildProjectGraphUsingContext(
 }
 
 function createContext(
-  projectsConfigurations: ProjectsConfigurations,
+  projects: Record<string, ProjectConfiguration>,
   nxJson: NxJsonConfiguration,
   fileMap: ProjectFileMap,
   filesToProcess: ProjectFileMap
 ): ProjectGraphProcessorContext {
-  const projects = Object.keys(projectsConfigurations.projects).reduce(
-    (map, projectName) => {
-      map[projectName] = {
-        ...projectsConfigurations.projects[projectName],
-      };
-      return map;
-    },
-    {} as Record<string, ProjectConfiguration>
-  );
```

Comment:
I think this should remain...

This clones the projects that we pass into plugins so that they don't mutate the ones that we pass into subsequent plugins

Reply:
added cloned back